src/ActionModal/utils/ActionManager.js

```diff
@@ -103,6 +103,12 @@ export default class ActionManager {
           txMessages.push(txMessage)
         })
       )
+      // HACK until this is refactored in a proper way
+      const claimAmountMessage = {
+        type: "/".concat(type),
+        value: { amount: transactionProperties.amounts }
+      }
+      txMessages.push(claimAmountMessage)
```

Comment:
lol I get it now. very hacky... ^^
you remember that it is a bad idea to mix data types. same goes for adding pseudo messages to an actual message array.

Reply:
Proposal:

extend `getSigner` in `signer.js` to return functions of the format `(signMessage, metaData) => ...` where `metaData = { claimableRewards }` then pass `metaData` to `signWithExtension` and send it to the extension. The extension now reads this prop and uses it to display this information.

Reply:
OK, so then I don't throw this into the messages... it feels weird to call this a metaData though. Especially when there is already a `txMetaData` already dancing around in ActionManager (can lead to confusion)

Reply:
yeah this whole thing needs to be heavily refactored. you can call it anything else. how about `displayingData`.

Reply:
Well, I will added it right there then. To `txMetadata`

Reply:
What about https://github.com/luniehq/lunie/pull/3736#issuecomment-601202588?